src/sentry/ingest/transaction_clusterer/datasource/redis.py

```diff
@@ -20,38 +20,60 @@
 SET_TTL = 24 * 60 * 60
 
 
+REDIS_KEY_PREFIX = "txnames:"
+
 add_to_set = redis.load_script("utils/sadd_capped.lua")
 
 
 def _get_redis_key(project: Project) -> str:
-    return f"txnames:o:{project.organization_id}:p:{project.id}"
+    return f"{REDIS_KEY_PREFIX}o:{project.organization_id}:p:{project.id}"
 
 
-def _get_redis_client() -> Any:
+def get_redis_client() -> Any:
     cluster_key = getattr(settings, "SENTRY_TRANSACTION_NAMES_REDIS_CLUSTER", "default")
     return redis.redis_clusters.get(cluster_key)
 
 
+def _get_all_keys() -> Iterable[str]:
+    client = get_redis_client()
+    return client.scan_iter(match=f"{REDIS_KEY_PREFIX}*")  # type: ignore
+
+
+def get_active_projects() -> Iterable[Project]:
+    """Scan redis for projects and fetch their db models"""
+    for key in _get_all_keys():
+        project_id = int(key.split(":")[-1])
+        yield Project.objects.get_from_cache(id=project_id)
+
+
 def _store_transaction_name(project: Project, transaction_name: str) -> None:
     with sentry_sdk.start_span(op="txcluster.store_transaction_name"):
-        client = _get_redis_client()
+        client = get_redis_client()
         redis_key = _get_redis_key(project)
         add_to_set(client, [redis_key], [transaction_name, MAX_SET_SIZE, SET_TTL])
 
 
 def get_transaction_names(project: Project) -> Iterable[str]:
     """Return all transaction names stored for the given project"""
-    client = _get_redis_client()
+    client = get_redis_client()
     redis_key = _get_redis_key(project)
 
     return client.sscan_iter(redis_key)  # type: ignore
 
 
 def record_transaction_name(project: Project, event: Event, **kwargs: Any) -> None:
-    source = (event.data.get("transaction_info") or {}).get("source")
-    if (
-        source == TRANSACTION_SOURCE
-        and event.transaction
-        and features.has("organizations:transaction-name-clusterer", project.organization)
+    transaction_info = event.data.get("transaction_info") or {}
+    transaction_name = event.transaction
+    source = transaction_info.get("source")
+    if transaction_name and features.has(
+        "organizations:transaction-name-clusterer", project.organization
```

Comment:
This check also exists before running the task, so there are two checks for each org. Should we relay on the other one, and assume at this point that we actually want to record the event?

Reply:
This check is executed in post-processing, while the other check is in the new periodical task. I would keep both checks because when you switch the feature flag to off, you want clustering to stop immediately, and not only after the redis sets expire.